image-download

```diff
@@ -236,11 +236,10 @@ def committed_target(image):
 def wait_lock(target):
     lockfile = os.path.join(os.path.dirname(target), ".lock." + os.path.basename(target))
 
-    # we need to keep the lock fd open throughout the entire runtime, so remember it in a global-scoped variable
-    wait_lock.f = open(lockfile, "w")
+    lock_fd = open(lockfile, "w")
```

Comment:
Please don't call this `fd` — it's not.  It's makes the `fd.close()` line below very confusing.

Reply:
Dropped that commit as you wanted to rework this.